src/runtime/virtcontainers/clh.go

```diff
@@ -395,8 +425,33 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 
 	// Create the VM memory config via the constructor to ensure default values are properly assigned
 	clh.vmconfig.Memory = chclient.NewMemoryConfig(int64((utils.MemUnit(clh.config.MemorySize) * utils.MiB).ToBytes()))
-	// shared memory should be enabled if using vhost-user(kata uses virtiofsd)
-	clh.vmconfig.Memory.Shared = func(b bool) *bool { return &b }(true)
+
+	if !clh.config.BootFromTemplate {
+		// shared memory should be enabled if using vhost-user(kata uses virtiofsd)
+		// or VM templating builds the first VM with file-backed memory
+		clh.vmconfig.Memory.Shared = func(b bool) *bool { return &b }(true)
+	}
+
+	// VM templating builds VM with file-backed memory
+	if clh.config.BootToBeTemplate || clh.config.BootFromTemplate {
+		clh.vmconfig.Memory.Size = 0 
+		// virtio-mem is the only way of resizing a memory zone
+		var hotplugMethod string = "VirtioMem"
+		clh.vmconfig.Memory.HotplugMethod = &hotplugMethod
+		memoryZoneConfig := chclient.NewMemoryZoneConfig("mem0", int64((utils.MemUnit(clh.config.MemorySize) * utils.MiB).ToBytes()))
+		memoryZoneConfig.File = &clh.config.MemoryPath
+		if clh.config.BootToBeTemplate {
+			err := os.Truncate(clh.config.MemoryPath, int64((utils.MemUnit(clh.config.MemorySize) * utils.MiB).ToBytes()))
+			if err != nil {
+				return err
+			}
+			memoryZoneConfig.Shared = func(b bool) *bool { return &b }(true)
```

Comment:
Thanks for your review。With the current implementations, VM templating will not work with file based memory (stand-alone) or virtiofs. This is because VM templating builds the first VM with file-backed memory and shared=on and the
subsequent ones with shared=off. virtio-fs always requires shared=on for memory. （I see this information on virtcontainers/qemu.go ）

Reply:
Thanks for your review。With the current implementations, VM templating will not work with file based memory (stand-alone) or virtiofs. This is because VM templating builds the first VM with file-backed memory and shared=on and the subsequent ones with shared=off. virtio-fs always requires shared=on for memory. （I see this information on virtcontainers/qemu.go ）

Reply:
Thanks for the context. Ping @devimc @fidencio @amshinde for more information.